pages/articles.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+    <div class="pa-5">
+        <div v-if="loading">
+            <Loading :active="loading"
+                :can-cancel="false"
+                :is-full-page="true"
+                color="#007bff"
+            />
+        </div>
+        <div v-else>
+            <v-row align="center" class="mx-0">
+                <v-col :cols="4" v-for="article, index in currentArticleList" :key="index">
+                    <ArticleCard
+                        :key="index"
+                        :articleData="article"
+                    />
+                </v-col>
+            </v-row>
+
+            <v-pagination
+                circle
+                v-model="currentPage"
+                :length="totalPages"
+                @input="onPageNumberClick"
+            ></v-pagination>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+// Library imports
+import { Vue, Component } from 'vue-property-decorator'
+import Loading from 'vue-loading-overlay'
+import 'vue-loading-overlay/dist/vue-loading.css'
+
+// Components imports
+import ArticleCard from '~/components/article_card.vue'
+
+// Utils and Interfaces imports
+import { API_KEY, APIS } from '~/utils/globals'
+import { ArticleObj } from '~/utils/interfaces'
+
+@Component({
+    components: {
+        ArticleCard,
+        Loading
+    }
+})
+export default class Articles extends Vue {
+
+    // Component data
+    pageSize: number = 10
+    currentPage: number = 1
+    allArticleList: ArticleObj[] = []
+    currentArticleList: ArticleObj[] = []
+    loading: boolean = true
+    $helper: any // To avoid Typescript errors related to type safety
+
+
+    // Computed Properties
+    get totalPages(): number {
+        return Math.ceil(this.allArticleList.length / this.pageSize)
+    }
+
+
+    // Methods
+    async getAllData() {
+        const { EVERYTHING } = APIS
+        const params = {
+            sources: "bbc-news",
+            apiKey: API_KEY
+        }
+        try {
+            const response = await this.$axios.get(EVERYTHING, {
+                params
+            })
```

Comment:
if you are using Axios plugIn you should have setted up all these in it's plugin/axios.ts file.

Reply:
Okay. But since it is a practice project I configured this on nuxt config file for the base url :smile: :sweat_smile: 
For practice To understand the plugins folder I create helper file as mentioned above :sweat_smile: